### ISSUE TITLE
How to create DEFAULT_TOKEN 

### ISSUE THREAD

**Issue by anhnt0212** [ktamas77/firebase-php#46]
Hello guys.
How do I get DEFAULT_TOKEN configuration ?

**Comment by greabock** [comment]
@anhnt0212  same problem. Have you any solution?

**Comment by parthjdabhi** [comment]
@ktamas77 @llbbl @brianpilati  @craigrusso can you help ?

**Comment by anhnt0212** [comment]
@greabock  i don't have :(

**Comment by anhnt0212** [comment]
Can you help me ?

**Comment by skizzo** [comment]
Chin up champs, in your firebase console, just go to the **project settings**, then to the tab **database** and then **copy the secret key**. Worked like a charm for me!

![firebasephpdefaulttoken](https://cloud.githubusercontent.com/assets/2892572/17928830/0ac038bc-69fd-11e6-8b4d-b7492edc6235.jpg)

**Comment by anhnt0212** [comment]
@skizzo tks a lot.let me check again :D

**Comment by martisj** [comment]
Does this database secret override all the security rules defined?

**Comment by anhnt0212** [comment]
@martisj i think it no

**Comment by codemonkeynorth** [comment]
I believe this is deprecated now? Or am i lookng at the wrong thing?

![image](https://cloud.githubusercontent.com/assets/20189985/20038341/db627834-a429-11e6-9efd-7e5de4c83413.png)

**Comment by quocthinh212** [comment]
![token](https://cloud.githubusercontent.com/assets/14103334/21080624/2f69e4c6-bfe6-11e6-9431-8732c32aa5ab.png)

**Comment by codemonkeynorth** [comment]
yes the secret still works but it is deprecated. you ought to be using Service Accounts now but this library does not yet support it

v1.x/2.x Firebase JWT token's  that @anhnt0212 is using are not the same as the secret.

you could try kreait's version instead, which supports the Service Accounts used by v3.x firebase

having looked at that codebase, it's not  simple so it's no small task that we are asking @ktamas77 to do to update this library to the new version by the way 

what are you actually trying to do @anhnt0212 ?

**Comment by codemonkeynorth** [comment]
ps if you use REST to access Firebase .json I believe authentication works in either of the following ways

...something.json?auth={secret} // DEFAULT_TOKEN in ktamas77's docs
...something.json?auth={JWT}
...something.json?access_token={v3_service_account_oauth_token_etc}

I believe the middle one is what the token generator library you are using provides @anhnt0212 

as you can see the token is passed in the querystring options for `auth`
https://github.com/ktamas77/firebase-php/blob/master/src/firebaseLib.php#L107

you could try creating a JWT token with **Firebase token generator** using your secret, and then pass that token into @ktamas77's library... if you don't want to pass your secret directy. I've not tried it though. it depends if it should be passed as auth= or access_token=
